sdetools/connector_plugin/docs/sde/connector_plugins/jira_plugin/jira_plugin.py

```
@@ -0,0 +1,65 @@
+    id = "JIRAConnector"
+    name = "JIRA Connector"
+    description = "Retrieves project information from a JIRA instance"
+
+    @property
+    def configuration_params(self):
+        server = ConfigurationParam(name="server",
+                                    description="Example: https://myserver.atlassian.net",
+                                    label="Server")
+
+        username = ConfigurationParam(name="username",
+                                      label="User Name")
+
+        password = ConfigurationParam(name="password",
+                                      label="Password",
+                                      param_type=ConfigurationParamType.MASKED_STRING)
+
+        return [server, username, password]
+
+    @property
+    def expected_custom_attributes(self):
+        jira_project_id = CustomAttribute(name="jira_project_id")
+        return [jira_project_id]
+
+    def fetch_project_data(self, id, configuration):
+        project_details = self.get_project_details(id, configuration)
+
+        if not project_details:
+            msg = "Cannot find project in JIRA. Make sure you are using the right key and credentials"
+            return {
+              "errors": [msg]
+            }
+
+        return {
+            'name': project_details['name'],
+            'description': project_details['description'],
+            'custom_attributes': {
+                'jira_project_id': project_details['id']
+            }
+        }
+
+    def get_project_details(self, id, configuration):
+        try:
+            req_url = "%s/rest/api/2/project/%s" % (configuration['server'], id)
+            request = urllib2.Request(req_url)
+            base64string = base64.encodestring('%s:%s' % (configuration['user'], configuration['pass'])).replace('\n', '')
+            request.add_header("Authorization", "Basic %s" % base64string)
+            result = urllib2.urlopen(request).read()
+            return json.loads(result)
+        except:
```

We shouldn't use catch all exceptions. It makes debugging harder, as it can sometimes mask errors. We should catch the exceptions we expect, and fail on the ones we don't.